phoenix/app/apiProducts/lib/apiProducts/index_product.ex

```diff
@@ -1,26 +1,34 @@
 defmodule ApiProducts.IndexProduct do
 
   import Tirexs.HTTP
-  alias ApiProducts.Catalog
 
-  def product_index({:ok, product}) do
+  def product_index(product) do
     product_json =
        %{
         id: product.id,
         name: product.name,
         sku: product.sku,
         description: product.description,
         qtd: product.qtd,
-        price: product.price
+        price: product.price,
         date: DateTime.to_iso8601(DateTime.utc_now())
       }
-  end
 
-  def products(product) do 
     put("/apiProducts/products/#{product_json.id}", product_json)
-  end 
+  end
 
-  def delete_product_by_index(product) do
+  def delete_product(product) do
     delete("/apiProducts/products/#{product.id}")
   end
+
+  def search_product(params) do
+    query = Enum.map_join(params, "*&", fn{k, v} -> "#{k}:#{v}" end)
```

Comment:
Faça um AND aqui. Se a pessoa passar um filtro como `name=teste&sku=123` e existir o name, mas não com esse sku, não é para retornar